l1-contracts/contracts/bridge/asset-tracker/GWAssetTracker.sol

```diff
@@ -138,6 +136,10 @@ contract GWAssetTracker is AssetTrackerBase, IGWAssetTracker {
         bytes32 _canonicalTxHash,
         BalanceChange calldata _balanceChange
     ) external onlyL2InteropCenter {
+        uint256 migrationNumber = _getChainMigrationNumber(_chainId);
+
+        /// We save the chainBalance for the previous migration number so that the chain balance can be migrated back to GW in case it was not migrated.
+        _getOrSaveChainBalance(_chainId, _balanceChange.assetId, migrationNumber - 1, false);
```

Comment:
if a chain tried to migrate, but failed, then this logic will not be correct. 

What's more if a chain tried to migrate and failed for the first time, `migrationNumber` is 0 and so it will underflow